fastai2/vision/utils.py

```diff
@@ -44,4 +44,35 @@ def verify_image(fn):
 # Cell
 def verify_images(fns):
     "Find images in `fns` that can't be opened"
-    return L(fns[i] for i,o in enumerate(parallel(verify_image, fns)) if not o)
\ No newline at end of file
+    return L(fns[i] for i,o in enumerate(parallel(verify_image, fns)) if not o)
+
+# Cell
+def resize_image(file, max_size:None, dest=None, n_channels=3, ext=None,
+                 img_format=None, resume=False, **kwargs ):
+    verify_image(file)
+    img = Image.open(file)
+    imgarr = np.array(img)
+    img_channels = 1 if len(imgarr.shape) == 2 else imgarr.shape[2]
+    if (max_size is not None and (img.height > max_size or img.width > max_size)) or img_channels != n_channels:
+        assert isinstance(dest, Path), "You should provide `dest` Path to save resized image"
+        dest_fname = dest/file.name
+        if ext is not None: dest_fname=dest_fname.with_suffix(ext)
+        if resume and os.path.isfile(dest_fname): return
+        if max_size is not None:
+            new_sz = resize_to(img, max_size)
+            img = img.resize(new_sz, resample=3)
+        if n_channels == 3: img = img.convert("RGB")
+        img.save(dest_fname, img_format, **kwargs)
+
+# Cell
+def resize_images(path, max_workers=4, max_size=None, recurse=False,
+                  dest=Path('.'), n_channels=3, ext=None, img_format=None,
+                  resume=None, **kwargs):
+    path = Path(path)
+    if resume is None and dest == '.': resume=False
```

Comment:
More like `resume= dest != Path('.')`? Otherwise there is no default behavior.

Reply:
I don't really get what is resume here, the idea is that a file is already there, it should not be copied, hein?

Reply:
Yes, since resizing can be a long process.